src/build-mounting-container.ts

```diff
@@ -0,0 +1,40 @@
+import { ElementOrSelector } from './types';
+
+/**
+ * Helper function for getting an element when given a string or element
+ * @param target a selector or the container
+ * @returns the container that we can mount on
+ */
+const getContainer = (target: ElementOrSelector): HTMLElement => {
+	// if the selector is a string, try to find the element,
+	// otherwise it's probably DOM that we should write directly to
+	if (typeof target === 'string') {
+		const selectedElement = document.querySelector(target);
+		if (selectedElement === null) {
+			throw new Error('Tram-One: could not find target, is the element on the page yet?');
+		}
+		return selectedElement;
+	} else {
+		return target;
+	}
+};
+
+/**
+ * Function to determine (or create) the element that we will mount our tram-one app onto
+ * @param target either a CSS selector, or HTMLElement to attach the component to.
+ * This elememnt should be initially empty.
+ *
+ * @returns the container, now with a div that tram-one can manage
+ */
+export default (target: ElementOrSelector): HTMLElement => {
```

Comment:
DocStrings no longer have types in them. These _should_ be inherited from the Types in the method definition / implementation.